Applications/Utils/FileConverter/MeshToRaster.cpp

```diff
@@ -0,0 +1,205 @@
+std::size_t getProjectedElement(std::vector<const MeshLib::Element*> const& elems,
+                                MeshLib::Node const& node)
+{
+   // std::vector<MeshLib::Element*> const& elems = mesh.getElements();
+    std::size_t const n_elems = elems.size();
+    for (std::size_t i = 0; i < n_elems; ++i)
+    {
+        if (elems[i]->getGeomType() == MeshLib::MeshElemType::LINE)
+            continue;
+        MeshLib::Node const& a = *(elems[i]->getNode(0));
+        MeshLib::Node const& b = *(elems[i]->getNode(1));
+        if (GeoLib::isRightOf(node, a, b))
+            continue;
+        MeshLib::Node const& c = *(elems[i]->getNode(2));
+        if (GeoLib::isRightOf(node, b, c))
+            continue;
+        if (elems[i]->getGeomType() == MeshLib::MeshElemType::TRIANGLE)
+        {
+            if (GeoLib::isRightOf(node, c, a))
+                continue;
+        }
+        if (elems[i]->getGeomType() == MeshLib::MeshElemType::QUAD)
+        {
+            MeshLib::Node const& d = *(elems[i]->getNode(3));
+            if (GeoLib::isRightOf(node, c, d))
+                continue;
+            if (GeoLib::isRightOf(node, d, a))
+                continue;
+        }
+        return elems[i]->getID();
+    }
+    return std::numeric_limits<size_t>::max();
+}
+
+/// Returns the z-coordinate of a point projected onto the plane defined by a
+/// mesh element.
+double getElevation(MeshLib::Element const& elem, MeshLib::Node const& node)
+{
+    MathLib::Vector3 const n = MeshLib::FaceRule::getSurfaceNormal(&elem).getNormalizedVector();
+    MeshLib::Node const orig = *elem.getNode(0);
+    MathLib::Point3d const v{{node[0] - orig[0], node[1] - orig[1], node[2] - orig[2]}};
+    double const dist = n[0] * v[0] + n[1] * v[1] + n[2] * v[2];
+    return node[2] - dist * n[2];
+}
+
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    TCLAP::CmdLine cmd(
+        "Mesh to raster converter.\n"
+        "Rasterises a 2D mesh, pixel values are set to the elevation of a "
+        "regular grid superimposed on the mesh. If no mesh element is located "
+        "beneath  a pixel it is set to NODATA.\n\n"
+        "OpenGeoSys-6 software, version " +
+            BaseLib::BuildInfo::git_describe +
+            ".\n"
+            "Copyright (c) 2012-2019, OpenGeoSys Community "
+            "(http://www.opengeosys.org)",
+        ' ', BaseLib::BuildInfo::git_describe);
+    TCLAP::ValueArg<std::string> input_arg(
+        "i", "input-file", "Mesh input file (*.vtu, *.msh)", true, "", "string");
+    cmd.add(input_arg);
+    TCLAP::ValueArg<std::string> output_arg(
+        "o", "output-file", "Raster output file (*.asc)", true, "", "string");
+    cmd.add(output_arg);
+    TCLAP::ValueArg<double> cell_arg(
+        "c", "cellsize", "edge length of raster cells in result", false, 1, "real");
+    cmd.add(cell_arg);
+
+    cmd.parse(argc, argv);
+
+    INFO("Rasterising mesh...");
+    if (!input_arg.isSet())
```

Comment:
Isn't tclap already testing this? The third last argument to the constructor of the argument is the requirement, isn't it?

Reply:
:heavy_check_mark: